components/channel_view/index.js

```diff
@@ -29,11 +31,16 @@ function mapStateToProps(state) {
     const config = getConfig(state);
     const enableTutorial = config.EnableTutorial === 'true';
     const tutorialStep = getInt(state, Preferences.TUTORIAL_STEP, getCurrentUserId(state), TutorialSteps.FINISHED);
+    const channel = getCurrentChannel(state);
+    const isReadOnly = channel && channel.name === Constants.DEFAULT_CHANNEL &&
+        !UserStore.isSystemAdminForCurrentUser() &&
+        config.ExperimentalTownSquareIsReadOnly === 'true';
 
     return {
         channelId,
         deactivatedChannel: getDeactivatedChannel(state, channelId),
         showTutorial: enableTutorial && tutorialStep <= TutorialSteps.INTRO_SCREENS,
+        isReadOnly,
```

Comment:
If we had a selector, would it be simpler if we just had the connected components grab that value from the store themselves when they need it instead of passing it up from here? There's a lot of layout components that have been changed by this PR just to pass around `isReadOnly` to their children.